pkg/cli/build.go

```diff
@@ -115,13 +116,13 @@ func cmdBuild() *cobra.Command {
 	return cmd
 }
 
-type configStuff struct {
+type dagConfig struct {
 	g    *dag.Graph
 	m    map[string]map[string]graph.Edge[string]
 	pkgs *dag.Packages
 }
 
-func walkConfigs(ctx context.Context, cfg *global) (*configStuff, error) {
+func walkConfigs(ctx context.Context, dir, pipelineDir string, extraKeys, extraRepos []string) (*dagConfig, error) {
```

Comment:
despite the name, `global` isn't actually global (its tied to `build`).

this is kind of a lazy workaround, but if there's a need I'm happy to refactor it better!

Reply:
I don't have a lot to offer on reasoning 😂 but was mostly curious because I tend to lean on the struct format rather than long argument lists because the former is easier to maintain (but may introduce boilerplate code so not a strong argument anyways).

Reply:
I take that back. I am now curious whether you think an interface would be a bad idea here to keep the previous structure.